WolvenKit.App/ViewModels/Documents/RedDocumentViewModel.cs

```diff
@@ -274,6 +284,42 @@ public CR2WFile GetFileFromDepotPathOrCache(CName depotPath)
             return Files[depotPath];
         }
 
+        public ICommand NewEmbeddedFileCommand { get; private set; }
+        private void ExecuteNewEmbeddedFile()
+        {
+            var existing = new ObservableCollection<string>(AppDomain.CurrentDomain.GetAssemblies().SelectMany(s => s.GetTypes()).Where(p => p.IsAssignableTo(typeof(CResource)) && p.IsClass).Select(x => x.Name));
+
+            var app = Locator.Current.GetService<AppViewModel>();
+            app.SetActiveDialog(new CreateClassDialogViewModel(existing, true)
+            {
+                DialogHandler = HandleEmbeddedFile
+            });
+        }
+
+        public void HandleEmbeddedFile(DialogViewModel sender)
+        {
+            var app = Locator.Current.GetService<AppViewModel>();
+            app.CloseDialogCommand.Execute(null);
+            if (sender != null)
+            {
+                var dvm = sender as CreateClassDialogViewModel;
```

Comment:
`as` might return `null`, maybe it is better to use `is CreateClassDialogViewModel dvm`.